### PR TITLE
Import ResponderEventPlugin from react-native

### PR DESCRIPTION
**Pull request by spicyj**
Two commits: the first is a direct import of the files from react-native, with the license headers updated; the second makes some minor updates so that it works again without the NodeHandle abstraction that we don't use (currently?) in core.

### CODE REVIEW

**Comment by spicyj** [comment]
(This helps us unfork.)

File: src/event/EventPropagators.js [hunk 1]
```diff
@@ -40,10 +38,9 @@ function listenerAtPhase(id, event, propagationPhase) {
  */
 function accumulateDirectionalDispatches(domID, upwards, event) {
   if (__DEV__) {
-    warning(
-      domID,
-      'Dispatching id must not be null'
-    );
+    if (!domID) {
+      throw new Error('Dispatching id must not be null');
+    }
```
Comment (zpao) [inline src/event/EventPropagators.js]: Didn't we intentionally change this to not throw to bring dev & prod behavior in line? (#2868)

Comment (spicyj) [inline src/event/EventPropagators.js]: Good catch. I'll revert this bit.

File: src/core/ReactInstanceHandles.js [hunk 2]
```diff
@@ -115,8 +115,7 @@ function getNextDescendantID(ancestorID, destinationID) {
   // Skip over the ancestor and the immediate separator. Traverse until we hit
   // another separator or we reach the end of `destinationID`.
   var start = ancestorID.length + SEPARATOR_LENGTH;
-  var i;
-  for (i = start; i < destinationID.length; i++) {
+  for (var i = start; i < destinationID.length; i++) {
```
Comment (zpao) [inline src/core/ReactInstanceHandles.js]: This should be failing lint I thought (for violating theoretical "block level" scoping for `i`)

Comment (spicyj) [inline src/core/ReactInstanceHandles.js]: I wish that's how the linter worked but I don't think it is.

**Comment by zpao** [comment]
Actually there are a bunch of lint failures…

**Comment by spicyj** [comment]
Oh! Right you are. I'll fix.

**Comment by spicyj** [comment]
(lint fixed)

**Comment by zpao** [comment]
:thumbsup:

**Pull request merged**